http.go

```diff
@@ -995,6 +995,34 @@ func (resp *Response) ReadLimitBody(r *bufio.Reader, maxBodySize int) error {
 	return nil
 }
 
+func (resp *Response) copyBody(wr io.Writer, rd io.Reader, maxBodySize int) error {
+	resp.resetSkipHeader()
+	r := bufio.NewReader(rd)
+	err := resp.Header.Read(r)
+	if err != nil {
+		return err
+	}
+	if resp.Header.StatusCode() == StatusContinue {
+		// Read the next response according to http://www.w3.org/Protocols/rfc2616/rfc2616-sec8.html .
+		if err = resp.Header.Read(r); err != nil {
+			return err
+		}
+	}
+	if !resp.mustSkipBody() {
+		n, t := int(0), int64(0)
+		if n = resp.Header.ContentLength(); maxBodySize > 0 && n > maxBodySize {
+			n = maxBodySize
+		}
+		t, err = io.CopyN(wr, rd, int64(n))
```

Comment:
I don't think this is going to work. Above `r` is a `bufio.Reader` which reads from `r` in blocks and buffers it in memory. So when reading the headers from `r` it might already read part of the body as well as a buffer. On this line you are reading from `rd` again which then won't include that part of the body anymore. I think this needs to read from `r` as well for this to always work.

Reply:
Also this would only work with a `Content-Length` set. For Chunked or Identity encoding this would just return the encoded body instead of decoding it like it's supposed to, see [readBody](https://github.com/erikdubbelboer/fasthttp/blob/d014e440c1557aa83ecdab57d7144245c650e0f0/http.go#L1581-L1584).

Reply:
Okay. I will fix and push it later.